libs/vkd3d/command.c

```diff
@@ -1337,9 +1337,18 @@ static void d3d12_shared_fence_dec_ref(struct d3d12_shared_fence *fence)
 {
     ULONG refcount_internal = InterlockedDecrement(&fence->refcount_internal);
     const struct vkd3d_vk_device_procs *vk_procs;
+    struct vkd3d_shared_fence_waiting_event *current, *e;
 
     if (!refcount_internal)
     {
+        vkd3d_atomic_uint64_store_explicit(&fence->stop, 1, vkd3d_memory_order_release);
+        pthread_join(fence->thread, NULL);
+
+        LIST_FOR_EACH_ENTRY_SAFE(current, e, &fence->events, struct vkd3d_shared_fence_waiting_event, entry)
+        {
+            vkd3d_free(current);
```

Comment:
Is this safe? Wouldn't current->next be de-referenced right after?

Reply:
The `LIST_FOR_EACH_ENTRY_SAFE` version copies the list to each step (hence why it has 2 ""iterators""). This is also done somewhere else in the code base, so I'm pretty sure it is safe.